core/board.py

```diff
@@ -43,9 +13,47 @@ def update(self, user_shot_choice):
                 '7': 6,
                 '8': 7,
                 '9': 8,
-                '10': 9
+                '10': 9,
+                }
+        columns = {
+                'A': 0,
+                'B': 1,
+                'C': 2,
+                'D': 3,
+                'E': 4,
+                'F': 5,
+                'G': 6,
+                'H': 7,
+                'I': 8,
+                'J': 9,
                 }
         user_letter = user_shot_choice[0]
         user_num = user_shot_choice[1:]
+        
+        self.state[rows[user_num]][columns[user_letter]] = 'Hit' if is_hit else 'Miss' 
+            
+    def add_to_board(self, all_ships, place, ship_orientation):
```

Comment:
this functions has a lot going on in it. Definitely violating SRP and pretty hard to follow.